app/src/main/kotlin/view/utils/EditDBWindow.kt

```
@@ -0,0 +1,165 @@
+package view.utils
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import model.io.sql.SQLDatabaseModule
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun EditDBWindow(DBType: String, onDismiss: () -> Unit) {
+    var showDialog by remember { mutableStateOf(false) }
+    val graphNamesSQL = remember { mutableStateOf(arrayListOf<Pair<Int, String>>()) }
+    var expanded by remember { mutableStateOf(false) }
+    var selectedGraphName by remember { mutableStateOf("") }
+    var selectedGraphID by remember { mutableStateOf(0)}
+    var updateGraphNames by remember { mutableStateOf(false) }
+    var graphNameToReplaceWith by remember { mutableStateOf("")}
+
+    if (DBType == "SQLite") {
```

❗️ What if this is false? The app will silently fail. I understand you've probably programmed this to be impossible but silent failures are not an option: you should either make them impossible at compile time (in this case, use a enum and handle all its values) or at least explicitly throw an `IllegalArgumentException`/`IllegalStateException` (or use `require()`/`check()` which is equivalent). This makes possible programming errors easier to notice.

---

I added my statement relying on the hope of "will be implemented in future", but at the end of deadline we haven't had all db implementations(